snapcraft/extractors/appstream.py

```diff
@@ -75,14 +77,20 @@ def extract(relpath: str, *, workdir: str) -> ExtractedMetadata:
     summary = _get_value_from_xml_element(dom, "summary")
     description = _get_value_from_xml_element(dom, "description")
 
-    icon = None
     node = dom.find("icon")
-    if node is not None and "type" in node.attrib and node.attrib["type"] == "local":
-        # TODO Currently we only support local icons.
+    icon = node.text.strip() if node is not None else None
```

Comment:
how about moving this code block for icon management closer to where you use it again after the desktop file parsing?
Maybe even consider renaming node to icon_node, node_type to icon_node_type OR even better, extract all this into its own method or better yet and Icon-like class

Reply:
Refactored to move node parsing into the icon extraction function, and cleaned up the logic a little bit there.